src/App.tsx

```diff
@@ -2,24 +1,39 @@
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const totalItems = items.length;
+
+  const from = (currentPage - 1) * Number(perPage) + 1;
+  const to =
+    currentPage * perPage > totalItems ? totalItems : currentPage * perPage;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {from} - {to} of {totalItems})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            value={perPage}
+            onChange={event => {
+              setPerPage(Number(event.target.value));
+              setCurrentPage(1);
+            }}
```

Comment:
When changing the items per page, you're setting the current page back to 1, which is good. However, consider the case when the current page is already 1; this would cause unnecessary state updates. To optimize, you could check if the current page is not 1 before setting it.